py-polars/polars/functions/aggregation/vertical.py

```diff
@@ -0,0 +1,535 @@
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING, Any, Iterable, overload
+
+import polars._reexport as pl
+import polars.functions as F
+from polars.utils.decorators import deprecated_alias
+from polars.utils.various import find_stacklevel
+
+if TYPE_CHECKING:
+    from polars import Expr, Series
+    from polars.type_aliases import IntoExpr, PythonLiteral
+
+
+@overload
+def all(exprs: Series) -> bool:  # type: ignore[misc]
+    ...
+
+
+@overload
+def all(
+    exprs: IntoExpr | Iterable[IntoExpr] | None = ..., *more_exprs: IntoExpr
+) -> Expr:
+    ...
+
+
+@deprecated_alias(columns="exprs")
+def all(
+    exprs: IntoExpr | Iterable[IntoExpr] | None = None, *more_exprs: IntoExpr
+) -> Expr | bool:
+    """
+    Either return an expression representing all columns, or evaluate a bitwise AND operation.
+
+    If no arguments are passed, this is an alias for ``pl.col("*")``.
+    If a single string is passed, this is an alias for ``pl.col(name).any()``.
+
+    If a single Series is passed, this is an alias for ``Series.any()``.
+    **This functionality is deprecated**.
+
+    Otherwise, this function computes the bitwise AND horizontally across multiple
+    columns.
+    **This functionality is deprecated**.
+
+    Parameters
+    ----------
+    exprs
+        Column(s) to use in the aggregation. Accepts expression input. Strings are
+        parsed as column names, other non-expression inputs are parsed as literals.
+    *more_exprs
+        Additional columns to use in the aggregation, specified as positional arguments.
+
+    Examples
+    --------
+    Selecting all columns.
+
+    >>> df = pl.DataFrame(
+    ...     {
+    ...         "a": [True, False, True],
+    ...         "b": [False, False, False],
+    ...     }
+    ... )
+    >>> df.select(pl.all().sum())
+    shape: (1, 2)
+    ┌─────┬─────┐
+    │ a   ┆ b   │
+    │ --- ┆ --- │
+    │ u32 ┆ u32 │
+    ╞═════╪═════╡
+    │ 2   ┆ 0   │
+    └─────┴─────┘
+
+    Evaluate bitwise AND for a column:
+
+    >>> df.select(pl.all("a"))
+    shape: (1, 1)
+    ┌───────┐
+    │ a     │
+    │ ---   │
+    │ bool  │
+    ╞═══════╡
+    │ false │
+    └───────┘
+
+    """  # noqa: W505
+    if not more_exprs:
+        if exprs is None:
+            return F.col("*")
```

Comment:
Calling `all()` without any inputs should trigger this code path and not throw any DeprecationWarnings.

Reply:
Hmm.. My IDE (jetbrains) claims I call a deprecated function. It indeed doesn't get a warning. Maybe the deprecated alias is too broad?